dataikuapi/govern/artifact_search_handler.py

```diff
@@ -0,0 +1,203 @@
+        if artifact_search_source_type == "blueprintVersions":
+            artifact_search_source["blueprintVersionIds"] = ids_list
+        if artifact_search_source_type == "artifacts":
+            artifact_search_source["artifactIds"] = ids_list
+
+        if artifact_filters_list is None:
+            artifact_filters_list = []
+        query = {"artifactFilters": [artifact_filter.build() for artifact_filter in artifact_filters_list]}
+        if sort_direction is not None or sort_column is not None:
+            query["artifactSearchSort"] = {"direction": sort_direction, "column": {"type": sort_column}}
+
+        return GovernArtifactSearchQuery(artifact_search_source, query, )
+
+    def build_request(self, query):
+        """
+        Create a new artifact search request and return the object that will be used to launch the request
+
+        :param :class:`dataikuapi.govern.artifact_search_handler.GovernArtifactSearchQuery` query: The query that will
+        be addressed during the search.
+        :returns The created artifact search request object
+        :rtype: :class:`dataikuapi.govern.artifact_search_handler.GovernArtifactSearchRequest`
+        """
+
+        return GovernArtifactSearchRequest(self.client, query)
+
+
+class GovernArtifactSearchQuery(object):
+    """
+    An object which represents a query
+    """
+
+    def __init__(self, artifact_search_source, query):
+        self.artifact_search_source = artifact_search_source
+        self.query = query
+
+    def build_query(self):
+        return {"artifactSearchSource": self.artifact_search_source,
+                "query": self.query}
+
+
+class GovernArtifactSearchRequest(object):
+    """
+    A search request object.
+    Do not create this directly, use :meth:`dataikuapi.govern.artifact_search_handler.build_query()` and then run the
+    query using :meth:`dataikuapi.govern.artifact_search_handler.GovernArtifactSearchRequest.perform_search()`
+    """
+
+    def __init__(self, client, query):
+        self.client = client
+        self.query = query
+
+    def perform_search(self, page_size=20, last_artifact_id=None):
+        """
+        Run the search request. Use page_size and last_artifact_id to get the paginated results.
+
+        :param int page_size: size of the result page, default value is set to 20.
+        :param str last_artifact_id: id of the last artifact. Useful to get the next page of result starting from a
+        specific id.
+        :returns The result of the search request. This dict contains a key "uiArtifacts" which is the list of the
+        results list. The dict contains a key "hasNextPage"  which value is boolean. If this value is set to true, it is
+        possible to navigate through the results using the parameters page_size and last_artifact_id.
+        :rtype: dict
+        """
+        body = self.query
+        if page_size != 20:
+            body["pageSize"] = page_size
```

Comment:
is page size an attribute directly on the query, it's more in an artifactpaginationsettings sub element?